lib/querystring.js

```diff
@@ -178,7 +179,7 @@ function qsEscape(str) {
     if (i < str.length)
       c2 = str.charCodeAt(i) & 0x3FF;
     else
-      throw new URIError('URI malformed');
+      throw new errors.TypeError('ERR_QUERYSTRING_URI_INVALID', 'URI malformed');
```

Comment:
The error type `URIError` should stay as it is. If that type does not yet exist in the `internal/errors` it has to be created.
Out of my perspective a better name  for the internal type would also be `ERR_INVALID_URI` (I am still thinking about if it would be a good idea to combine this with the existing `ERR_INVALID_URL` type or not) and the error message itself should be moved in the internal errors as static value.

Reply:
As a temporary workaround this could be done with:
```js
else {
  const e = new URIError('URI malformed');
  e.code = ERR_INVALID_URL;
  throw e;
}
```

Reply:
Since this is the only place in `/lib/` where `URIError` is used, I'm not sure it should not be changed...
/cc @jasnell

Reply:
This recommendation is bad out of my perspective. It is not similar to the `internal/errors` by just adding the error code. It is neither a `NodeError`, nor would it receive any further changes in case something changes in the `internal/errors`.
I would also like to keep the error type as `URIError` even if it is the only occurrence in lib. It is a single line that has to be added to the `internal/errors` to add that type and use it from then on.

Reply:
Thank for your comments
but I am not fully understand you
do you mean to add "URIError" to module.exports in internal/errors?
e.g: 
```js
module.exports = exports = {
  message,
  URIError, 
  ....
```
or using the exported Error with URIError
e.g: 
```js
throw new errors.Error(URIError)
```